File: src/game/item/Key.java
package game.item;

import game.TextureManager;
import game.Vector;
import game.creature.Creature;
import game.creature.Player;

public class Key extends Item {
	public Key(Vector pos) {
		super(pos, TextureManager.KEY);
		color = new float[] {1, 1, 0};
	}
	
	@Override
	public boolean pickedUpBy(Creature c) {
		try {
			Player p = (Player) c;
			p.hasKey = true;
			return true;
		} catch (ClassCastException e) {
			return false;
		}
	}
}

File: ts/game/item/Key.ts
import { TextureManager } from '../TextureManager.js';
import { Vector } from '../Vector.js';
import { Creature } from '../creature/Creature.js';
import { Player } from '../creature/Player.js';
import { Item } from './Item.js';

export class Key extends Item {
  constructor(pos: Vector) {
    super(pos, TextureManager.KEY);
  }

  pickedUpBy(c: Creature): boolean {
    if (c instanceof Player) {
      c.hasKey = true;
      return true;
    }
    return false;
  }
}
